### ISSUE TITLE
GUI freezes while adding/removing scripts through the API

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#4302]
Issue reported in OWASP ZAP User Group: https://groups.google.com/d/topic/zaproxy-users/8SKAQnXy6z4/discussion

Steps to reproduce the issue:
1. Run ZAP (with GUI);
2. Add and remove (several times) a script through the API;
3. Note that ZAP GUI froze, a thread dump indicates that the EDT and an API request are deadlocked:
```
Found one Java-level deadlock:
=============================
"AWT-EventQueue-0":
  waiting to lock monitor 0x00007fa9c4250e80 (object 0x0000000085843b60, a sun.awt.im.InputMethodContext),
  which is held by "ZAP-ProxyThread-6116"
"ZAP-ProxyThread-6116":
  waiting to lock monitor 0x00007fa9c4360f00 (object 0x0000000083bb4d58, a java.awt.Component$AWTTreeLock),
  which is held by "AWT-EventQueue-0"

Java stack information for the threads listed above:
===================================================
"AWT-EventQueue-0":
        at sun.awt.im.InputContext.removeNotify(java.desktop@10-ea/InputContext.java:613)
        - waiting to lock <0x0000000085843b60> (a sun.awt.im.InputMethodContext)
        at java.awt.Component.removeNotify(java.desktop@10-ea/Component.java:7107)
        - locked <0x0000000083bb4d58> (a java.awt.Component$AWTTreeLock)
        at java.awt.Container.removeNotify(java.desktop@10-ea/Container.java:2844)
        - locked <0x0000000083bb4d58> (a java.awt.Component$AWTTreeLock)
        at javax.swing.JComponent.removeNotify(java.desktop@10-ea/JComponent.java:4801)
        at java.awt.Container.removeAll(java.desktop@10-ea/Container.java:1319)
        - locked <0x0000000083bb4d58> (a java.awt.Component$AWTTreeLock)
        at javax.swing.plaf.synth.SynthTreeUI.paint(java.desktop@10-ea/SynthTreeUI.java:426)
        at javax.swing.plaf.synth.SynthTreeUI.update(java.desktop@10-ea/SynthTreeUI.java:264)
        at javax.swing.JComponent.paintComponent(java.desktop@10-ea/JComponent.java:797)
        at javax.swing.JComponent.paint(java.desktop@10-ea/JComponent.java:1074)
        at javax.swing.JComponent.paintToOffscreen(java.desktop@10-ea/JComponent.java:5255)
        at javax.swing.RepaintManager$PaintManager.paintDoubleBufferedImpl(java.desktop@10-ea/RepaintManager.java:1633)
        at javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(java.desktop@10-ea/RepaintManager.java:1608)
        at javax.swing.RepaintManager$PaintManager.paint(java.desktop@10-ea/RepaintManager.java:1546)
        at javax.swing.BufferStrategyPaintManager.paint(java.desktop@10-ea/BufferStrategyPaintManager.java:262)
        at javax.swing.RepaintManager.paint(java.desktop@10-ea/RepaintManager.java:1313)
        at javax.swing.JComponent._paintImmediately(java.desktop@10-ea/JComponent.java:5203)
        at javax.swing.JComponent.paintImmediately(java.desktop@10-ea/JComponent.java:5013)
        at javax.swing.RepaintManager$4.run(java.desktop@10-ea/RepaintManager.java:857)
        at javax.swing.RepaintManager$4.run(java.desktop@10-ea/RepaintManager.java:840)
        at java.security.AccessController.doPrivileged(java.base@10-ea/Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(java.base@10-ea/ProtectionDomain.java:87)
        at javax.swing.RepaintManager.paintDirtyRegions(java.desktop@10-ea/RepaintManager.java:840)
        at javax.swing.RepaintManager.paintDirtyRegions(java.desktop@10-ea/RepaintManager.java:815)
        at javax.swing.RepaintManager.prePaintDirtyRegions(java.desktop@10-ea/RepaintManager.java:764)
        at javax.swing.RepaintManager.access$1200(java.desktop@10-ea/RepaintManager.java:69)
        at javax.swing.RepaintManager$ProcessingRunnable.run(java.desktop@10-ea/RepaintManager.java:1880)
        at java.awt.event.InvocationEvent.dispatch(java.desktop@10-ea/InvocationEvent.java:313)
        at java.awt.EventQueue.dispatchEventImpl(java.desktop@10-ea/EventQueue.java:770)
        at java.awt.EventQueue.access$600(java.desktop@10-ea/EventQueue.java:97)
        at java.awt.EventQueue$4.run(java.desktop@10-ea/EventQueue.java:721)
        at java.awt.EventQueue$4.run(java.desktop@10-ea/EventQueue.java:715)
        at java.security.AccessController.doPrivileged(java.base@10-ea/Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(java.base@10-ea/ProtectionDomain.java:87)
        at java.awt.EventQueue.dispatchEvent(java.desktop@10-ea/EventQueue.java:740)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(java.desktop@10-ea/EventDispatchThread.java:203)
        at java.awt.EventDispatchThread.pumpEventsForFilter(java.desktop@10-ea/EventDispatchThread.java:124)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(java.desktop@10-ea/EventDispatchThread.java:113)
        at java.awt.EventDispatchThread.pumpEvents(java.desktop@10-ea/EventDispatchThread.java:109)
        at java.awt.EventDispatchThread.pumpEvents(java.desktop@10-ea/EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.run(java.desktop@10-ea/EventDispatchThread.java:90)
"ZAP-ProxyThread-6116":
        at sun.awt.im.InputContext.focusLost(java.desktop@10-ea/InputContext.java:453)
        - waiting to lock <0x0000000083bb4d58> (a java.awt.Component$AWTTreeLock)
        at sun.awt.im.InputContext.removeNotify(java.desktop@10-ea/InputContext.java:627)
        - locked <0x0000000085843b60> (a sun.awt.im.InputMethodContext)
        at java.awt.Component.enableInputMethods(java.desktop@10-ea/Component.java:1600)
        at javax.swing.text.JTextComponent.setEditable(java.desktop@10-ea/JTextComponent.java:1816)
        at org.zaproxy.zap.extension.scripts.CommandPanel.setEditable(CommandPanel.java:126)
        at org.zaproxy.zap.extension.scripts.ConsolePanel.clearScript(ConsolePanel.java:319)
        at org.zaproxy.zap.extension.scripts.ExtensionScriptsUI.scriptRemoved(ExtensionScriptsUI.java:582)
        at org.zaproxy.zap.extension.script.ExtensionScript.removeScript(ExtensionScript.java:602)
        at org.zaproxy.zap.extension.script.ScriptAPI.handleApiAction(ScriptAPI.java:189)
        at org.zaproxy.zap.extension.api.API.handleApiRequest(API.java:431)
        at org.parosproxy.paros.core.proxy.ProxyThread.processHttp(ProxyThread.java:456)
        at org.parosproxy.paros.core.proxy.ProxyThread.run(ProxyThread.java:317)
        at java.lang.Thread.run(java.base@10-ea/Thread.java:844)
```
(VisualVM  thread dump.)

Remarks:
It might not happens always but happens often (specially if the script being removed is selected).

Add-on
Script Console version 23 (and older).

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#1300.

**Comment by thc202** [comment]
Released in version 24 of Script Console add-on.
